src/tools/SelectionTool.js

```diff
@@ -0,0 +1,92 @@
+import { createRef } from "preact";
+import Tool from "./Tool";
+import { createSvgFromPoints } from "../utils";
+
+class SelectionTool extends Tool {
+  constructor(model) {
+    super(model);
+    this.toolName = 'SELECT';
+  }
+
+  reset() {
+    super.reset();
+    this.model.selection = null;
+  }
+
+  onPress( action ) {
+    if (!this.model.controlsactive) {
+      return;
+    }
+    if (this.model.activityInProgress) {
+      return;
+    }
+
+    super.onPress();
+    this.model.activityInProgress = true;
+
+    const { x, y } = action;
+    const ref = createRef();
+
+    this.model.selection = [
+      'path',
+      {
+        ref: ref,
+        fill: 'none',
+        points: [],
+        start: {'x': x, 'y': y},
+        d: `M${x} ${y}`,
+        stroke: 'lightgray',
+        'stroke-width': '2',
+        'stroke-linejoin': 'round',
+        'stroke-linecap': 'round',
+        'stroke-dasharray': '5,3', 
+        'vector-effect': 'non-scaling-stroke',
+      }, 'selection',
+    ];
+  }
+
+  onMove( action ) {
+    if (!this.model.activityInProgress) {
+      return;
+    }
+
+    const { x, y } = action;
+    if (!this.model.selection) {
+        return;
```

Comment:
when could this happen, that model.selection is null but we have a selection event

Reply:
This was to be safe while creating and testing the tool, but shouldn't happen. Removed.